### PR TITLE
[RFR][developer] project architecture

### PR DESCRIPTION
**Pull request by ndufreche**
backbee/backbee-standard#338

### CODE REVIEW

File: doc/developper-documentation/project_architecture.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+- `Data` Folder where every uploaded file are stored. Against cache and log folder, data can moved out of the project by customizing `services.yml`
+
+- `Layouts` Layouts template are stored in this folder.
+
+- `Listener` In this folder your Event Listeners will be automatically auto-loaded.
+
+- `Resources` In this folder you put your statics files like images, JS scripts and CSS.
+
+- `Template` Folder where you put your helpers into `helpers` subfolder, partials into `partials` subfolder and ClassContent templates into `scripts` subfolder.
```
Comment (fkroockmann) [inline doc/developper-documentation/project_architecture.md]: Templates

File: doc/developper-documentation/project_architecture.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+- `Translation` Your translation catalog will automatically auto-loaded in this folder.
+
+### Context and repository folder overriding
+
+In BackBee repository folder can be contextualized. You use an contextualized repository when you start your application, it's the first parameter in construct method.
+
+```php
+<?php
+$app = new \BackBee\Standard\Application('newContext');
+// ...
```
Comment (fkroockmann) [inline doc/developper-documentation/project_architecture.md]: I think we need to explain how you can use this context.

Comment (fkroockmann) [inline doc/developper-documentation/project_architecture.md]: By creating a folder 'newContext' into Config folder with overrided config file

File: doc/developper-documentation/project_architecture.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+## Project Architecture
+
+### Global Architecture
+
+Global Architecture is build by default by BackBee, it's compose the folders:
+- `bundle` optional folder, create it only if you want develop your own bundle. By default the bundle is installed into the vendor folder
+- `cache` optional  and can be placed out of the project and is location have to be specified into `services.yml` into the `Config` folder.
+- `log` same as cache folder
+- `public` folder it is where you will find the entry point to your application
```
Comment (crouillon) [inline doc/developper-documentation/project_architecture.md]: I'd prefer "the document root of your website", strictly talking, the entry point of the application is the class Application.php

**Comment by crouillon** [comment]
Could add the configuration overriding rules and the resolving resource request rules?

**Comment by crouillon** [comment]
:+1:

**Comment by mickaelandrieu** [comment]
thank you @ndufreche

**Pull request merged**